BENCHMARKING.md

```diff
@@ -0,0 +1,178 @@
+# Benchmarking Wild
+
+## Benchmarking against other linkers
+
+If you decide to benchmark Wild against other linkers, in order to make it a fair comparison, you
+should ensure that the other linkers aren't doing work on something that Wild doesn't support. In
+particular:
+
+* Wild defaults to `--gc-sections`, so for a fair comparison, that should be passed to all the linkers.
+* Wild defaults to `-z now`, so best to pass that to all linkers.
+
+There might be other flags that speed up the other linkers by letting them avoid some work that
+they're currently doing. If you know of such flags, please let me know.
+
+## How to benchmark
+
+### Preparing the "run-with" files
+
+For benchmarking the linker, it's preferable to run just the linker, not the whole build process.
+
+The way to do that is by capturing the linker invocation so that it can be rerun. Wild has a built-in way to do that.
+
+You can benchmark linking of either a debug or a release build of a crate, this depends on what comparisons you wish to
+make, or what change in wild you want to quantify.
+
+Follow-these steps:
+
+* Chose the crate that you wish to use in your benchmark, clone it, `cd` into it's root directory and make sure it
+  builds with `cargo build` (for a rust project)
+    * Examples: [`ripgrep`](https://github.com/BurntSushi/ripgrep.git)
+* Clean the build using `cargo clean`
+* To force the build of your chosen crate to link using wild, we have a couple of options:
+    * Prefix the cargo build command with `RUSTFLAGS="-Clinker=clang -Clink-arg=--ld-path=wild"`
+    * Modify (or add) the `.cargo/config.toml` file in your chosen crate (example for `ripgrep`)
+
+```toml
+  [target.x86_64-unknown-linux-gnu]
+linker = "/usr/bin/clang"
+
+rustflags = [
+    "-C", "link-arg=--ld-path=wild"
+]
+```
+
+* Make sure that you have a version of wild in your `$PATH` so that it will be used (try `which wild` to check)
+* Run `WILD_SAVE_BASE=/tmp/wild/ripgrep cargo build` in the crate's root directory (include `RUSTFLAGS` as above
+  if you have chosen that method)
+* You will get a few numbered subdirectories in `/tmp/wild/ripgrep` for different binaries being linked in the build
+  process (such as build scripts and multiple binaries in a crate)
+* Usually the last numbered subdirectory will be the final link stage of the binary (TODO clarify that)
```

Comment:
The numbers will be allocated in whatever order the binaries are built. So if you have one main binary, that will get built after all the build scripts and proc macros. If you have multiple binaries, potentially any of them could be last.